Core/include/Acts/Surfaces/Surface.hpp

```diff
@@ -252,12 +252,15 @@ class Surface : public virtual GeometryObject,
   /// @param position global position to be evaludated
   /// @param direction global momentum direction (required for line-type surfaces)
   /// @param boundaryTolerance BoundaryTolerance directive for this onSurface check
+  /// @param tolerance optional tolerance within which a point is considered
+  /// valid on surface
   ///
   /// @return boolean indication if operation was successful
-  bool isOnSurface(const GeometryContext& gctx, const Vector3& position,
-                   const Vector3& direction,
-                   const BoundaryTolerance& boundaryTolerance =
-                       BoundaryTolerance::None()) const;
+  bool isOnSurface(
+      const GeometryContext& gctx, const Vector3& position,
+      const Vector3& direction,
+      const BoundaryTolerance& boundaryTolerance = BoundaryTolerance::None(),
+      double tolerance = s_onSurfaceTolerance) const;
```

Comment:
Could be renamed to `onSurfaceTolerance`, to more clearly seperate it from the `boundaryTolerance`

Reply:
I am also quite unhappy with `tolerance`. I think right not it is consistent in a way that we always use `tolerance` for surface tolerance and `boundaryTolerance` for the other. I would propose that we streamline this in a different PR all together